prometheus_client/bridge/newrelic.py

```diff
@@ -0,0 +1,27 @@
+from __future__ import absolute_import
+
+import newrelic.agent
+
+from .. import core
+
+
+@newrelic.agent.data_source_factory(name='Prometheus')
+class PrometheusDataSource(object):
+    def __init__(self, settings, environ):
+        self.registry = settings.get('registry', core.REGISTRY)
+        self.prefix = settings.get('prefix', '')
+
+    def __call__(self):
+        prefixstr = ''
+        if self.prefix:
+            prefixstr = self.prefix + '.'
+
+        for metric in self.registry.collect():
+            for name, labels, value in metric.samples:
+                if labels:
+                    labelstr = '.' + '.'.join(['{0}.{1}'.format(k, v) for k, v in sorted(labels.items())])
```

Comment:
https://docs.newrelic.com/docs/plugins/plugin-developer-resources/developer-reference/metric-naming-reference indicates that / * | [ ] also need special handling.

Having the labels as metric attributes using /  would seem a better fit of the data model than a dotted string.